### DISCUSSION TITLE
Replacing the "with" key in quests.json for another key name

### DISCUSSION THREAD

**Discussion by stphn** [TarkovTracker/tarkovdata#249] [Schema Changes]
Hello there,

In [quests.json](https://github.com/TarkovTracker/tarkovdata/blob/master/quests.json) the `with` key is a reserved word and reserved word cannot be used as a shorthand property (React etc..).
Maybe a key name like `using` could solve some issues.

Thanks for reading and sharing your thoughts

**Comment by thaddeus** [comment]
Hey! Good catch. I hadn't considered that previously.  I would not be opposed to switching to `using`. In the meantime, are you able to access the properties through another syntax like `myObjective['with']`?

@Razzmatazzz for awareness

**Reply by stphn** [reply]
Thanks,

It's sunny out there, I will try that syntax proposition and come back to you.

I'm glad I can help

**Reply by stphn** [reply]
This is how I use it:

maybe there is a workaround?

``` javascript
export const Objective = ({
    type,
    target,
    tool,
    hint,
    have,
    number,
    location,
    with, // causing issue
    id,
    gps,
    map,
}: QuestObjective) => {
    

    return (
        <div className='objective'>
            {type}
            {tool}
            {hint}
            {with} {/* causing issue */}
            {have}
            {gps?.floor}
            {gps?.leftPercent}
            {gps?.topPercent}
            {map}
        </div>
    )
}
```

**Reply by thaddeus** [reply]
Sorry I am a TypeScript noob so I probably won't be able to provide great advice here. Are you able to do something like 

```
export const Objective = ({
    paramType: type,
    paramTarget: target,
    paramTool: tool,
    paramWith: with,
}: QuestObjective)
```

or

```
export const Objective = ({
    type as paramType,
    target as paramTarget,
    tool as paramTool,
    with as paramWith,
}: QuestObjective)
```

**Reply by stphn** [reply]
Thanks for your answer,

I was on the go so I couldn't reply earlier and the answer was:

``` typescript
export const Objective = ({
    type,
    target,
    tool,
    with: using,
}: QuestObjective)

```
Much easier and in case you really consider changing `with` for `using` the code is ready :) 

Thank for the help anyways I will follow any updates.

p.s: you definitely should try Typescript. I will soon provide you all the Types in a PR in case someone need them.

**Reply by thaddeus** [reply]
Thanks for the follow-up! Yeah, that PR would be very welcome! I do want to try getting into TypeScript. I'm just always in a rush trying to finish things. My next major update I'll have to make sure I start using it.
